packages/design-system/src/components/dropDown/test/dropDown.js

```diff
@@ -250,137 +250,121 @@ describe('DropDown <DropDown />', () => {
-
-//   await renderWithProviders(
-//     <DropDown
-//       emptyText={'No options available'}
-//       dropDownLabel={'label'}
-//       options={basicDropDownOptions}
-//       onMenuItemClick={onClickMock}
-//       selectedValue={basicDropDownOptions[0].value}
-//     />
-//   );
-//   const selectButton = screen.getByRole('button');
-//   fireEvent.click(selectButton);
-
-//   const menu = screen.getByRole('listbox');
-//   expect(menu).toBeInTheDocument();
-
-//   // focus first element in menu
-//   act(() => {
-//     fireEvent.keyDown(menu, {
-//       key: 'ArrowDown',
-//     });
-//   });
-
-//   // focus second element in menu
-//   act(() => {
-//     fireEvent.keyDown(menu, {
-//       key: 'ArrowDown',
-//     });
-//   });
-
-//   act(() => {
-//     fireEvent.keyDown(menu, { key: 'Enter' });
-//   });
-
-//   // The second item in the menu.
-//   // first prop we get back is the event
-//   expect(onClickMock).toHaveBeenCalledWith(
-//     expect.anything(),
-//     basicDropDownOptions[2].value
-//   );
-
-//   expect(onClickMock).toHaveBeenCalledTimes(1);
-
-//   await waitFor(() => expect(menu).not.toBeInTheDocument());
-// });
-
-// it('should trigger onDismissMenu when esc key is pressed', async () => {
-//   const wrapper = await renderWithProviders(
-//     <DropDown
-//       dropDownLabel={'label'}
-//       options={basicDropDownOptions}
-//       selectedValue={basicDropDownOptions[1].value}
-//     />
-//   );
-//   const select = wrapper.getByRole('button');
-//   fireEvent.click(select);
-
-//   const menu = wrapper.queryByRole('listbox');
-//   expect(menu).toBeInTheDocument();
-
-//   act(() => {
-//     fireEvent.keyDown(menu, {
-//       key: 'Escape',
-//     });
-//   });
-
-//   await waitFor(() => expect(menu).not.toBeInTheDocument());
-// });
+  describe('keyboard interactions', () => {
+    it('should trigger onMenuItemClick when using the down arrow plus enter key and keep menu open when isKeepMenuOpenOnSelection is true', async () => {
+      const onClickMock = jest.fn();
+      await renderWithProviders(
+        <DropDown
+          emptyText={'No options available'}
+          dropDownLabel={'label'}
+          isKeepMenuOpenOnSelection
+          options={basicDropDownOptions}
+          onMenuItemClick={onClickMock}
+          selectedValue={null}
+        />
+      );
+      const selectButton = screen.getByRole('button');
+      fireEvent.click(selectButton);
+
+      // wait for debounced callback to allow a select click handler to process
+      jest.runOnlyPendingTimers();
+
+      const menu = screen.getByRole('listbox');
+      expect(menu).toBeInTheDocument();
+
+      const menuItems = screen.getAllByRole('option');
+      expect(menuItems).toHaveLength(12);
+
+      // first element in menu should have focus
+      expect(menuItems[0]).toHaveFocus();
+
+      // focus second element in menu
+      fireEvent.keyDown(menuItems[1], { key: 'ArrowDown', which: 40 });
```

Comment:
Very subtle but the fix for these tests was to include `which: {keycode}` so that the `MouseTrap` picked up the event.

Reply:
Awesome find.